src/test/kotlin/org/rust/lang/core/type/RsExpressionTypeInferenceTest.kt

```diff
@@ -1,6 +1,9 @@
 package org.rust.lang.core.type
 
 import org.junit.ComparisonFailure
+import org.junit.Test
+import org.rust.lang.core.types.types.RustFloatType
+import org.rust.lang.core.types.types.RustIntegerType
 
 class RsExpressionTypeInferenceTest : RsTypificationTestBase() {
```

Comment:
Let's also add an integration tests, that will check that this works with real `::std::ops`. 

I think a test in `org.rust.lang.core.resolve.RsStdlibResolveTest` like this one

```Kotlin
    fun `test resolve operators`() = stubOnlyResolve("""
    //- main.rs
        fn main() {
            let x = String::new();
            let y = x + "World!";
            y.as_bytes()
        }       //^ ...libcollections/string.rs
    """)
```

should work. 

I think it fails currently for two reasons:

* it switches from stubs to AST, which is probably not that important
* the traits in `::std::ops` are generic, like `impl Add<&str> for String`, and this breaks  `.filter { it.traitRef?.text == traitName }`